cpp/src/arrow/type.h

```diff
@@ -1604,13 +1605,26 @@ class ARROW_EXPORT FieldRef {
 // ----------------------------------------------------------------------
 // Schema
 
+enum class Endianness {
+  LITTLE = 0,
+  BIG = 1,
+#if ARROW_LITTLE_ENDIAN
+  NATIVE = LITTLE
+#else
+  NATIVE = BIG
+#endif
+};
```

Comment:
@wesm Good point. To have an endianness flag in the Schema causes an inefficiency that passes this property as an argument.   
Do you want me to add an endian flag to `DataType` at https://github.com/apache/arrow/blob/master/cpp/src/arrow/type.h#L176 in this PR?
cc @pitrou 

For my reference, there is an API in Numpy at https://numpy.org/doc/stable/reference/generated/numpy.dtype.byteorder.html.

Reply:
@wesm Or, would it be possible to create a small PR to add an endian field to `DataType`? Then, I will continue to work for this PR after merging your PR.

Reply:
I will defer to @pitrou or @emkornfield on what to do -- I don't have a lot of bandwidth this month to help with writing PRs

Reply:
I see.

Reply:
@kiszk Indeed, I think we'll have to bit the bullet and add an endian field to `DataType`. Be sure to update other dependent methods (fingerprint computation, equality, tostring...) and add tests.

Reply:
@pitrou I see. It will be another set of PRs. This PR will not address to add an endian field to `DataType`.

After the set of PRs will be available, another PR for this feature will support endian fields in `DataType`.